modules/navier_stokes/src/actions/INSAction.C

```diff
@@ -0,0 +1,928 @@
+                                   RealVectorValue(0, 0, 0),
+                                   "The initial velocity, assumed constant everywhere");
+
+  params.addParamNamesToGroup(
+      "equation_type block gravity dynamic_viscosity_name density_name boussinesq_approximation "
+      "reference_temperature_name thermal_expansion_name",
+      "Base");
+  params.addParamNamesToGroup("use_ad laplace integrate_p_by_parts convective_term supg pspg alpha",
+                              "WeakFormControl");
+  params.addParamNamesToGroup("velocity_boundary velocity_function pressure_pinned_node "
+                              "no_bc_boundary pressure_boundary pressure_function",
+                              "BoundaryCondition");
+  params.addParamNamesToGroup(
+      "family order pressure_scaling velocity_scaling initial_pressure initial_velocity",
+      "Variable");
+  params.addParamNamesToGroup(
+      "add_temperature_equation temperature_variable temperature_scaling initial_temperature "
+      "thermal_conductivity_name specific_heat_name natural_temperature_boundary "
+      "fixed_temperature_boundary temperature_function",
+      "Temperature");
+  return params;
+}
+
+INSAction::INSAction(InputParameters parameters)
+  : Action(parameters),
+    _type(getParam<MooseEnum>("equation_type")),
+    _blocks(getParam<std::vector<SubdomainName>>("block")),
+    _velocity_boundary(getParam<std::vector<BoundaryName>>("velocity_boundary")),
+    _velocity_function(getParam<std::vector<FunctionName>>("velocity_function")),
+    _pressure_boundary(getParam<std::vector<BoundaryName>>("pressure_boundary")),
+    _pressure_function(getParam<std::vector<FunctionName>>("pressure_function")),
+    _no_bc_boundary(getParam<std::vector<BoundaryName>>("no_bc_boundary")),
+    _has_pinned_node(isParamValid("pressure_pinned_node")),
+    _pinned_node("ins_pinned_node"),
+    _fixed_temperature_boundary(getParam<std::vector<BoundaryName>>("fixed_temperature_boundary")),
+    _temperature_function(getParam<std::vector<FunctionName>>("temperature_function")),
+    _fe_type(Utility::string_to_enum<Order>(getParam<MooseEnum>("order")),
+             Utility::string_to_enum<FEFamily>(getParam<MooseEnum>("family"))),
+    _use_ad(getParam<bool>("use_ad")),
+    _temperature_variable_name(getParam<VariableName>("temperature_variable"))
+{
+  if (_pressure_function.size() != _pressure_boundary.size())
+    paramError("pressure_function",
+               "Size is not the same as the number of boundaries in 'pressure_boundary'");
+  if (_temperature_function.size() != _fixed_temperature_boundary.size())
+    paramError("temperature_function",
+               "Size is not the same as the number of boundaries in 'fixed_temperature_boundary'");
+  if (_use_ad)
+  {
+    if (parameters.isParamSetByUser("convective_term"))
+      mooseWarning("'convective_term' is ignored for AD");
+    if (!getParam<bool>("laplace"))
+      mooseError("'laplace' must be true for AD");
```

Comment:
Perhaps since you started working on this, we merged #15272, so AD now has support for the traction form

Reply:
Looks like the main NS markdown file needs to be modified to reflect this. I was reading it and though traction form is not available for AD. I will remove this and update the index.md file.